### ISSUE TITLE
One-click triaging

### ISSUE THREAD

**Issue by jnraine** [github/janky#249]
**tl;dr Let's make triaging flakes the choice that saves a developer time.**

When a build fails due to a flake candidate, the developer has two options:

1. **Triage the flake.** This benefits the organization by ensuring that others don't experience red builds but it takes time out of a developer's day.
2. **Click rebuild.** This takes less time and effort than triaging and is likely to have the same result for a developer: a green build after about ten minutes.

This is a tiny-scale example of the [tragedy of the commons](https://www.youtube.com/watch?v=CxC161GvMPc). It's in my best interest to click rebuild so that's often what I'll do. But "optimizing for the self in the short-term isn't optimal for anyone in the long-term."

### Problems with triaging

There are a few things that discourage developers from triaging:

- They must learn or be taught how Janky works.
- They must find an owner, which can be difficult and time consuming.
- They must still rebuild to unblock themselves.

### How to make it better

We'll address these directly by simplifying the triaging steps, automatically finding an owner, and removing the need to rebuild after triaging.

Here's how it will work:

1. **Your build fails** with one flake candidate ([like this](https://janky.githubapp.com/25772758/output)). It's failed over 50 times and delayed 10 trains, so the **[Triage now with 1-click](https://upload.wikimedia.org/wikipedia/commons/8/81/Amazon_1-Click_option.png)** button appears.
3. **You click the button.** An issue is created and assigned to the responsible person (i.e., not you).
4. **Your build goes green.**

That's it.

You can deploy now. Or add the PR to a train. Or send your code for review.

### TODOs

- [ ] Add additional failure metadata
  - [ ] Test last modified by
  - [ ] Test file last modified by
  - [ ] Source file last modified by
  - [ ] Codeowner
- [ ] Add additional information to triage issues
  - [ ] Add "why this is assigned to you" to issue description
  - [ ] Add flake candidate impact to issue description
- [ ] Add **Triage now with 1-click** button to "high impact" flakes
- [ ] Generate list of assignees from test failures
- [ ] Reprocess red/green status of affected builds after triaging a flake
